### DISCUSSION TITLE
How to pass record ID from resources page to custom pages and get the ID value

### DISCUSSION THREAD

**Discussion by MarJose123** [filamentphp/filament#4261] [Help] [answered]
How I can do the logic, as need to add custom action from the resource page and that action button will redirected to a custom page and auto-populate the select field under the custom page.

**Comment by danharrin** [comment]
If you pass the record ID as a query string parameter, you can fill the select default with `default(request('query_param_name'))`

**Reply by MarJose123** [reply]
is this will work for this kind of url `http://projects.test/portal/1` ?

**Reply by danharrin** [reply]
If you add that as a route parameter, yes? But if you just generate the route with query parameters then it's easier

**Reply by MarJose123** [reply]
> If you pass the record ID as a query string parameter, you can fill the select default with `default(request('query_param_name'))`

I tried this on a select field that but it not populating with the value from the parameter
```php
Select::make('Resident_name')
                    ->id('resident_name')
                    ->label('Resident\'s Full Name')
                    ->options(Residents::all()->pluck('FullName', 'id'))
                    ->searchable()
                    ->default(request('record'))
                    ->required()
                    ->reactive()
```

**Reply by danharrin** [reply]
Edit or create page? How are you passing the parameter into the route when you generate the URL?

**Reply by MarJose123** [reply]
> Edit or create page? How are you passing the parameter into the route when you generate the URL?

From the resources table, I've created a custom action button to redirect to a custom page with a parameter of record with the value of resident ID from the resident table DB.

code snippet resources table:
```php
 public static function table(Table $table): Table
    {
        return $table
            ->columns([

                 //.....

            ])
            ->filters([
               //...
            ])
            ->actions([
             //......
                Tables\Actions\Action::make('Certificate')
                    ->icon('heroicon-o-document-text')
                    ->color('success')
                    ->url(fn (Residents $record):string => Certification::getUrl(['record' => $record])),
            ])
            ->bulkActions([
              // .....
            ])
            ->headerActions([
                FilamentExportHeaderAction::make('export')
                    ->fileName('eBMS-Residents_Exported '.Carbon::now()->toDayDateTimeString())
                    ->disableAdditionalColumns()
                    ->withHiddenColumns()
                    ->disablePreview()
            ]);
    }
```

**Reply by danharrin** [reply]
You didn't pass the custom page route name in - `Certification::getUrl('page_name', ['record' => $record])`

**Reply by MarJose123** [reply]
> You didn't pass the custom page route name in - `Certification::getUrl('page_name', ['record' => $record])`

Anyway I solve it by my own :) thanks for the support and response.
I've used the `mount()` function to fill the form from query parameter since my custom page is not linked or generated from this command `php artisan make:filament-page SortUsers --resource=UserResource --type=custom`

**Comment by MarJose123** [comment, marked as answer]
I've used the `mount()` function to fill the form from query parameter

**Reply by ramiropiro** [reply]
how do you get the record var on mount section?

**Reply by MarJose123** [reply]
> how do you get the record var on mount section?

I used the `request()` to get the parameter value, then get the record by searching on the model using the parameter value provided.

**Comment by BOFOII** [comment]
just add  public YourModelClassTypeData $record;
